File: unix/memory-management/playground.js
let obj = {name : "kofi"}
let num = 700

function modify(object, number ){
    object.name = "Dylan"
    number = 1000
}

console.log(obj)
console.log(num)
modify(obj, num)
console.log("new obj value", obj)
console.log("new num value", num)

File: unix/memory-management/playground.c
#include <stdio.h>

struct Object
{
    /* data */
    const char *name;  //string
};

// function accepts 1. struct pointer to the struct Object, and 2. int pointer to number
void modify(struct Object *o, int *n){
    // Deferencing the value
  (*o).name = "Dylan";

   *n = 1000;
}

int main() {
    struct Object obj;
    obj.name = "Joe";
    int num = 700;

    printf("Before modification\n");
    printf("name%s\n", obj.name);
    printf("number %d\n", num);

    //modify
    modify(&obj, &num);

    printf("After modification\n");
    printf("name%s\n", obj.name);
    printf("number %d\n", num);
}